semtypes/src/main/java/io/ballerina/semtype/Range.java

```diff
@@ -0,0 +1,167 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.semtype;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Long.MAX_VALUE;
+import static java.lang.Long.MIN_VALUE;
+
+/**
+ * Int Range node.
+ *
+ * @since 2.0.0
+ */
+public class Range implements RangeHolder {
+    public final long min;
+    public final long max;
+
+    public Range(long min, long max) {
+        this.min = min;
+        this.max = max;
+    }
+
+    public static Range[] rangeListUnion(Range[] v1, Range[] v2) {
```

Comment:
These functions are only used in IntOps right, then shall we move them to IntOps and make `Range` just a data class.